parquet-variant/src/builder.rs

```diff
@@ -237,18 +237,37 @@ impl ValueBuffer {
 struct MetadataBuilder {
     // Field names -- field_ids are assigned in insert order
     field_names: IndexSet<String>,
+
+    // flag that checks if field names by insertion order are also lexicographically sorted
+    is_sorted: bool,
 }
 
 impl MetadataBuilder {
     /// Upsert field name to dictionary, return its ID
     fn upsert_field_name(&mut self, field_name: &str) -> u32 {
-        let (id, _) = self.field_names.insert_full(field_name.to_string());
+        let (id, new_entry) = self.field_names.insert_full(field_name.to_string());
+
+        if new_entry {
+            let n = self.num_field_names();
+
+            self.is_sorted =
+                n == 1 || self.is_sorted & (self.field_names[n - 2] < self.field_names[n - 1]);
```

Comment:
```suggestion
            self.is_sorted =
                n == 1 || self.is_sorted && (self.field_names[n - 2] < self.field_names[n - 1]);
```
(`&&` has short-circuit behavior and avoids the string comparison when we already broke sorting)

Reply:
Also, a code comment might help here, there's a lot to unpack:
* An empty dictionary is unsorted (ambiguous in spec but required by interop tests)
* A single-entry dictionary is trivially sorted
* Otherwise, an already-sorted dictionary becomes unsorted if the new entry breaks order